mopro-core/src/middleware/gpu_exploration/mod.rs

```diff
@@ -0,0 +1,86 @@
+use ark_bn254::{Fr as ScalarField, G1Affine as GAffine, G1Projective as G};
+use ark_ec::VariableBaseMSM;
+use ark_std::{error::Error, UniformRand};
+
+// For benchmarking
+use jemalloc_ctl::{epoch, stats};
+use std::time::{Duration, Instant};
+
+#[derive(Debug, Clone)]
+pub struct BenchmarkResult {
+    pub num_msm: u32,
+    pub avg_processing_time: f64,
+    pub total_processing_time: f64,
+    pub allocated_memory: f64,
+}
+
+#[global_allocator]
+static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
```

Comment:
Would you mind adding a comment for why we add this?

Reply:
After experiments, I discovered that this code will have the allocator record all the dynamic memory allocations. However, we only need to focus on processes related to MSM. I removed them to make the benchmarks more accurate.

Reply:
The new result shows that memory allocated for single MSM is $\approx{}0.05$ MiB

![msm benchmarks of time and memory usage](https://github.com/oskarth/mopro/assets/60170228/3bd2cc86-96d5-4e89-857b-ccbbd7635b3b)